source/dagger/gameplay/team_game/brawler_controller.cpp

```diff
@@ -27,88 +30,73 @@ void BrawlerControllerSystem::SpinUp()
 void BrawlerControllerSystem::Run()
 {
 
-    Engine::Registry().view<InputReceiver, Sprite, Animator, BrawlerCharacter>().each(
-        [](const InputReceiver input_, Sprite& sprite_, Animator& animator_, BrawlerCharacter& char_)
+    Engine::Registry().view<Physics,InputReceiver, Sprite, Animator, BrawlerCharacter>().each(
+        [](Physics& physics_,const InputReceiver input_, Sprite& sprite_, Animator& animator_, BrawlerCharacter& char_)
         {
+
             Float32 run = input_.values.at("run");
             Float32 jump = input_.values.at("jump");
-            Float32 attack = input_.values.at("light");
-            
-            if(!char_.jump || char_.attack){
-                if (char_.fall)
+           // Float32 attack = input_.values.at("light");
+
+            if (run != 0) {
+                sprite_.scale.x = run;
+                physics_.velocity.x += sprite_.scale.x* char_.speed;
+                char_.run = true;
+            }
+            /*if(!char_.jump )
```

Comment:
Why keep such a large comment? It's confusing to review, this way.

Reply:
Oh sorry forgot to delete that.